### PR TITLE
Add pagination to Usage by User Agent Report

### PR DESCRIPTION
**Pull request by Abhyastamita**
- Added pagination to Usage by User Agent Report
- Updated release notes

### CODE REVIEW

File: code/web/services/Admin/UsageByUserAgent.php [hunk 1]
```diff
@@ -18,8 +18,16 @@ function launch() {
 
 		$this->loadDates();
 
+		$page = isset($_REQUEST['page']) ? $_REQUEST['page'] : 1;
+		$pageSize = isset($_REQUEST['pageSize']) ? $_REQUEST['pageSize'] : 5; // to adjust number of items listed on a page
```
Comment (mdnoble73) [inline code/web/services/Admin/UsageByUserAgent.php]: Let's make the default page size 30 to match the selections in the menu.

File: code/web/services/Admin/UsageByUserAgent.php [hunk 2]
```diff
@@ -18,8 +18,16 @@ function launch() {
...
+		$interface->assign('recordsPerPage', $pageSize);
+		$interface->assign('page', $page);
+
 		$userAgent = new UserAgent();
 		$userAgent->orderBy('userAgent');
+		$total = $userAgent->count();
+		$userAgent->limit(($page - 1) * $pageSize, $pageSize);
+		$userAgent->find();
 		$allUserAgents = $userAgent->fetchAll('id', 'userAgent');
```
Comment (mdnoble73) [inline code/web/services/Admin/UsageByUserAgent.php]: The fetchAll statement redoes the search from the previous statement.  This just needs to be switched to a loop through the results and store the values appropriately.

**Review by mdnoble73 — CHANGES_REQUESTED** [review]

File: code/web/services/Admin/UsageByUserAgent.php (hunk 2 above)
Comment (Abhyastamita) [reply code/web/services/Admin/UsageByUserAgent.php]: Thanks Mark!  This is fixed now.

File: code/web/services/Admin/UsageByUserAgent.php (hunk 1 above)
Comment (Abhyastamita) [reply code/web/services/Admin/UsageByUserAgent.php]: Thanks!  I changed it to 30.

**Review by mdnoble73 — APPROVED** [review]

**Pull request merged**